libs/verify-desktop-environmnent.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Ensure the build toolchains are set up correctly for desktop builds.
+#
+# This file should be used via `./libs/verify-desktop-environment.sh`.
+
+set -e
+
+if [[ ! -f "$(pwd)/libs/build-all.sh" ]]; then
+  echo "ERROR: bootstrap-desktop.sh should be run from the root directory of the repo"
+  exit 1
+fi
+
+"$(pwd)/libs/bootstrap-common.sh"
+
+if [[ "$(uname -s)" == "Darwin" ]]; then
+  APPSERVICES_PLATFORM_DIR="$(pwd)/libs/desktop/darwin"
+else
+  APPSERVICES_PLATFORM_DIR="$(pwd)/libs/desktop/linux-x86-64"
+fi
+LIBS_SQLCIPHER_DIR="${APPSERVICES_PLATFORM_DIR}/sqlcipher"
+LIBS_NSS_DIR="${APPSERVICES_PLATFORM_DIR}/nss"
+
+if [[ ! -d "${LIBS_SQLCIPHER_DIR}" ]] || [[ ! -d "${LIBS_NSS_DIR}" ]]; then
+  pushd libs
+  ./build-all.sh desktop
+  popd
+fi;
+
+if [[ -z "${SQLCIPHER_LIB_DIR}" ]]; then
+  echo "Looks like the SQLCIPHER_LIB_DIR env variable is not set."
+  echo "We will then rely on sqlcipher begin installed system wide."
+  echo -n "Checking if it's present... "
+  if ! echo "int main() {}" | gcc -x c -o /dev/null - -lsqlcipher 2> /dev/null; then
+    echo ""
+    echo "Please install sqlcipher system wide."
+    echo " Linux: apt-get install -y sqlcipher"
```

Comment:
We require a much newer sqlcipher than the one that's in `apt` (which is about 4 years old). You should instead suggest building `libs` from source and setting the env vars in their shell's rc file.

Along these lines, it might be worth filing a bug that we should be checking the output of `sqlcipher --version` if possible.